main.tf

```diff
@@ -19,22 +19,22 @@ locals {
 
       resources = lookup(obj, "resources", null) == null ? {} : {
         for j, res in obj.resources :
-        "${var.trusted_profile_name}-${j}-resources" => res
+        "${obj.name}-${j}-resources" => res
       }
 
       resource_attributes = lookup(obj, "resource_attributes", null) == null ? {} : {
         for j, res_attr in obj.resource_attributes :
-        "${var.trusted_profile_name}-${j}-resource-attributes" => res_attr
+        "${obj.name}-${j}-resource-attributes" => res_attr
       }
 
       resource_tags = lookup(obj, "resource_tags", null) == null ? {} : {
         for j, res_tags in obj.resource_tags :
-        "${var.trusted_profile_name}-${j}-resource-tags" => res_tags
+        "${obj.name}-${j}-resource-tags" => res_tags
```

Comment:
instead of using ${j} here, use the value of resource_tags.name (its already required). Add validation to ensure its unique within the list